### PR TITLE
[bugfix] Fix stopping network update in parallel at max steps.

### PR DESCRIPTION
**Pull request by blattms**
We did only break in the loop for rank 0 and not the other ones. Hence all other processes kept iterating beyond the maximum number of allowed iterations. This lead to hard to find crashes because of non-matching MPI communication.

### CODE REVIEW

**Comment by blattms** [comment]
I will probably selfmerge as this shoud be uncontroversial.

**Comment by blattms** [comment]
jenkins build this please

**Review by GitPaean — APPROVED** [review]
Thanks for fixing this. It is a bug that I have to see is a little not that obvious.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1122,9 +1122,12 @@ namespace Opm {
...
                     updateWellControlsAndNetworkIteration(mandatory_network_balance, relax_network_balance, dt,local_deferredLogger);
             ++network_update_iteration;
-            if (terminal_output_ && (network_update_iteration >= max_iteration) ) {
-                local_deferredLogger.info("maximum of " + std::to_string(max_iteration) + " iterations has been used, we stop the network update now, "
-                                                                                          "the simulation will continue with unconvergeed network results");
+
+            if (network_update_iteration >= max_iteration ) {
+                if (terminal_output_) {
+                    local_deferredLogger.info("maximum of " + std::to_string(max_iteration) + " iterations has been used, we stop the network update now, "
+                                              "the simulation will continue with unconvergeed network results");
```
Comment (akva2) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:1129]: fix the converged while at it?

Comment (blattms) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: done

**Pull request merged**